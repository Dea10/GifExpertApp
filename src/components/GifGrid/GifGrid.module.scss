.grid {
    display: flex;
    flex-direction: row;
    padding-inline-start: 0px;
}