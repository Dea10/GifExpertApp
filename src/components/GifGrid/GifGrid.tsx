import React from 'react';
import useFetchGifs from '../../hooks/useFetchGifs';
import GifGridItem from '../GifGridItem/GifGridItem';
import styles from './GifGrid.module.scss';

type GifGridProps = {
    category: string;
}

const GifGrid = ({ category }: GifGridProps) => {

    const { data: images, loading } = useFetchGifs( category );

    return (
        <div>
        <h3>{category}</h3>
            {loading && 'loading...'}
            <ol className={ styles.grid }>
                {
                    images.map(img => (
                        <GifGridItem
                            key={ img.id }
                            { ...img } // rest parameter
                        />
                    ))
                }
            </ol>
        </div>
    );
}

export default GifGrid;