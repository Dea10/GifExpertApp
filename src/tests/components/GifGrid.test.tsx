import React from 'react';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid/GifGrid";

describe('GifGrid tests', () => {
    const category = 'xmen';
    const wrapper = shallow(<GifGrid category={ category } />)
    
    test('should render without errors', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('', () => {
        
    });
});