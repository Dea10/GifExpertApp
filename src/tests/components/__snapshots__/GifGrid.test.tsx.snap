// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should render without errors 1`] = `
<div>
  <h3>
    xmen
  </h3>
  loading...
  <ol
    className="grid"
  />
</div>
`;
